### PR TITLE
feat: export buttons for json and text format

### PR DESCRIPTION
**Pull request by piyushjha0409**
Created a generate text function and a button to export it.
Added a button for downloading in json format as well. :)

### CODE REVIEW

File: src/pages/stats/[login].tsx [hunk 1]
```diff
@@ -80,35 +151,43 @@ export default function Stats() {
...
               {contributions.nodes.map(({ pullRequest }) => (
-                <span key={pullRequest.id}>
+                <span key={pullRequest.id} id="pr-content">
                   {pullRequest.title} - {pullRequest.state}
                   <br />
                 </span>
               ))}
               <br />
             </div>
           ))}
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:180]: You can now replace all of this with generateText to make sure it is the exact same output

```suggestion
          {generateText()}
```

File: src/pages/stats/[login].tsx [hunk 2]
```diff
@@ -80,35 +151,43 @@ export default function Stats() {
...
+          <button className="bg-blue-500 p-2 m-1 rounded hover:bg-blue-900" onClick={exportText}>Export as Text</button>
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:181]: Can we put this button before the text? Same as you did for the JSON :)

File: src/pages/stats/[login].tsx [hunk 3]
```diff
@@ -19,6 +19,77 @@ export default function Stats() {
...
+            const prTitle = contribution.pullRequest.title;
+            const prState = contribution.pullRequest.state;
+            text += `- Pull Request: ${prTitle}\n`;
+            text += `  ID: ${prId}\n`;
+            text += `  State: ${prState}\n`;
+        }
+        
+        text += "\n";
+    }
+    console.log(text)
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:87]: Can you remove this `console.log`?

File: src/pages/stats/[login].tsx [hunk 4]
```diff
@@ -80,35 +151,43 @@ export default function Stats() {
       {isLoading ? (
         <div>Loading...</div>
       ) : format === "cards" ? (
+        <div>
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:154]: This duplicate `div` can be removed

File: src/pages/stats/[login].tsx [hunk 5]
```diff
@@ -2,7 +2,7 @@ import { RepositoryContributionsCard } from "@/components";
 import { useGitHubPullRequests } from "@/hooks";
 import { useSession } from "next-auth/react";
 import { useRouter } from "next/router";
-import { useState } from "react";
+import { useEffect, useState } from "react";
```
Comment (Balastrong) [inline src/pages/stats/[login].tsx:5]: useEffect is no longer used, can be removed

**Review by Balastrong — CHANGES_REQUESTED** [review]
I added a few comments but the overall logic seems correct! :)

Also please make sure to format the file (you can use `npm run format:fix` in your terminal) and restore the `.env.example` file :)

**Comment by piyushjha0409** [comment]
> I added a few comments but the overall logic seems correct! :)
> 
> Also please make sure to format the file (you can use `npm run format:fix` in your terminal) and restore the `.env.example` file :)

Sure

File: src/pages/stats/[login].tsx (hunk 1 above)
Comment (piyushjha0409) [reply src/pages/stats/[login].tsx:180]: image

**Comment by piyushjha0409** [comment]
I have done the changes, check it once!

**Review by Balastrong — APPROVED** [review]
LGTM! Thank you again! 🚀

**Pull request merged**